pyocd/gdbserver/gdbserver.py

```diff
@@ -152,6 +152,9 @@ def __init__(self, session, core=None, server_listening_callback=None):
         # Read back bound port in case auto-assigned (port 0)
         self.port = self.abstract_socket.port
 
+        # Coarse grain lock to synchronize SWO with other activity
+        self.lock = threading.Lock()
```

Comment:
The `DebugProbe` classes have a lock that should be used instead of creating a new one, along with `lock()` and `unlock()` methods.

Reply:
The `DebugProbe` class locks are too fine-grained, unfortunately.